### PR TITLE
Add Meta Boxes for Post Types

### PR DESCRIPTION
**Pull request by jesseeproductions**
- Adds all meta boxes and fields for the 3 post types. 

Artifact:
-Session Metaboxes: https://cloudup.com/cWjZG3QEQrx
-Speaker Metaboxes: https://cloudup.com/iD2CFCPfRgK
-Sponsor Metaboxes: https://cloudup.com/iYoOHF-xH6B

_Ref:_ [TEC-4882](https://theeventscalendar.atlassian.net/browse/TEC-4882)

[TEC-4882]: https://theeventscalendar.atlassian.net/browse/TEC-4882?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Conference/Plugin.php [hunk 1]
```diff
@@ -240,6 +240,7 @@ protected function register_autoloader() {
 		// Load Composer autoload and strauss autoloader.
 		require_once dirname( CONFERENCE_SCHEDULE_FILE ) . '/vendor/vendor-prefixed/autoload.php';
 		require_once dirname( CONFERENCE_SCHEDULE_FILE ) . '/vendor/autoload.php';
+		require_once dirname( CONFERENCE_SCHEDULE_FILE ) . '/vendor/cmb2/cmb2/init.php';
```
Comment (jesseeproductions) [inline src/Conference/Plugin.php:243]: - I changed to load cmb2 through composer so it is in the vendor directort.
- The file is directly loaded per their documentation. 
- CMB2 changes the main class name with each version update. 
- It also has checks if the class exists.

File: src/Conference/Admin/Meta/Session.php [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+	/**
+	 * Saves post session details.
+	 *
+	 * @since TBD
+	 *
+	 * @param int     $post_id Post ID.
+	 * @param WP_Post $post    Post object.
+	 */
+	public function save_post_session( $post_id, $post ) {
+		if ( wp_is_post_revision( $post_id ) || $post->post_type != 'wpcs_session' ) {
```
Comment (bordoni) [inline src/Conference/Admin/Meta/Session.php]: Do we have a constant that stores the name of the post type?

File: src/Conference/Admin/Meta/Session.php [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+			return;
+		}
+
+		if ( isset( $_POST['wpcs-meta-speakers-list-nonce'] ) && wp_verify_nonce( $_POST['wpcs-meta-speakers-list-nonce'], 'edit-speakers-list' ) && current_user_can( 'edit_post', $post_id ) ) {
+
+			// Update the text box as is for backwards compatibility.
+			$speakers = sanitize_text_field( $_POST['wpcs-speakers-list'] );
```
Comment (bordoni) [inline src/Conference/Admin/Meta/Session.php]: Use `Arr::get()`

File: src/Conference/Admin/Meta/Session.php [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+			update_post_meta( $post_id, '_conference_session_speakers', $speakers );
+		}
+
+		if ( isset( $_POST['wpcs-meta-session-info'] ) && wp_verify_nonce( $_POST['wpcs-meta-session-info'], 'edit-session-info' ) ) {
+
+			// Update session time
+			$session_time = strtotime( sprintf( '%s %d:%02d %s', sanitize_text_field( $_POST['wpcs-session-date'] ), absint( $_POST['wpcs-session-hour'] ), absint( $_POST['wpcs-session-minutes'] ), 'am' == $_POST['wpcs-session-meridiem'] ? 'am' : 'pm' ) );
```
Comment (bordoni) [inline src/Conference/Admin/Meta/Session.php]: Use `Arr::get()` for all the `$_POST` variables

File: src/Conference/Admin/Meta/Session.php [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+			update_post_meta( $post_id, '_wpcs_session_time', $session_time );
+
+			// Update session end time
+			$session_end_time = strtotime( sprintf( '%s %d:%02d %s', sanitize_text_field( $_POST['wpcs-session-date'] ), absint( $_POST['wpcs-session-end-hour'] ), absint( $_POST['wpcs-session-end-minutes'] ), 'am' == $_POST['wpcs-session-end-meridiem'] ? 'am' : 'pm' ) );
```
Comment (bordoni) [inline src/Conference/Admin/Meta/Session.php]: Use `Arr::get()` for all the `$_POST` variables

File: src/Conference/Admin/Meta/Session.php [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+			update_post_meta( $post_id, '_wpcs_session_end_time', $session_end_time );
+
+			// Update session type
+			$session_type = sanitize_text_field( $_POST['wpcs-session-type'] );
```
Comment (bordoni) [inline src/Conference/Admin/Meta/Session.php]: Use `Arr::get()` for all the `$_POST` variables

File: src/Conference/Admin/Meta/Session.php [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+			if ( ! in_array( $session_type, array( 'session', 'custom', 'mainstage' ) ) ) {
```
Comment (bordoni) [inline src/Conference/Admin/Meta/Session.php]:
```suggestion
			if ( ! in_array( $session_type, [ 'session', 'custom', 'mainstage' ], true ) ) {
```

File: src/Conference/Admin/Meta/Session.php [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+	/**
+	 * Adds the session information meta box.
+	 *
+	 * @since TBD
+	 */
+	public function session_metabox() {
+		$cmb = new_cmb2_box( [
+			'id'           => 'wpcs_session_metabox',
+			'title'        => _x( 'Session Information', 'Metabox title', 'wp-conference-schedule' ),
+			'object_types' => [ 'wpcs_session' ], // Post type
```
Comment (bordoni) [inline src/Conference/Admin/Meta/Session.php]: Use a constant with the slug of the CPT.

File: src/Conference/Admin/Meta/Session.php [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+
+		/**
+		 * Filters the speaker meta field in the session information meta box.
+		 *
+		 * @since TBD
+		 *
+		 * @param object $cmb CMB2 box object.
+		 */
+		if ( has_filter( 'wpcs_filter_session_speaker_meta_field' ) ) {
+			$cmb = apply_filters( 'wpcs_filter_session_speaker_meta_field', $cmb );
```
Comment (bordoni) [inline src/Conference/Admin/Meta/Session.php:91]: Docblock needs to be right above the filter.

File: src/Conference/Admin/Meta/Session.php [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+	 * Adds meta boxes for the session post type.
+	 *
+	 * @since TBD
+	 */
+	public function add_meta_boxes() {
+		add_meta_box(
+			'session-info',
+			_x( 'Session Info', 'Session Metabox Name.','wp-conference-schedule' ),
+			[ $this, 'metabox_session_info' ],
+			'wpcs_session',
```
Comment (bordoni) [inline src/Conference/Admin/Meta/Session.php]: Use constant.

File: src/Conference/Admin/Meta/Speaker.php [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+			'attributes' => [
+				'data-conditional-id'    => 'wpcsp_session_speaker_display',
+				'data-conditional-value' => 'typed'
+			]
+		] );
+
+		// Fetch sponsors
+		$args     = [
+			'numberposts' => - 1,
+			'post_type'   => 'wpcsp_sponsor',
```
Comment (bordoni) [inline src/Conference/Admin/Meta/Speaker.php]: constant.

**Review by bordoni — CHANGES_REQUESTED** [review]

File: src/Conference/Admin/Meta/Speaker.php (hunk 11 above)
Comment (jesseeproductions) [reply src/Conference/Admin/Meta/Speaker.php]: I made the updates, let me know if there is anything else. Thanks.

**Review by bordoni — APPROVED** [review]

**Pull request merged**